index.d.ts

```diff
@@ -0,0 +1,67 @@
+type IColorValue = string;
+
+// color1
+
+type IColor1Names = "black" | "white";
+type IColor1 = IColorValue;
+type IColors1 = { [name in IColor1Names]: IColor1 };
+
+// color2
+
+type IColor2Shades =
+  | "50"
+  | "100"
+  | "200"
+  | "300"
+  | "400"
+  | "500"
+  | "600"
+  | "700"
+  | "800"
+  | "900";
+type IColor2Names = "blueGrey" | "brown" | "grey";
+type IColor2 = { [shade in IColor2Shades]: IColorValue };
+type IColors2 = { [name in IColor2Names]: IColor2 };
+
+// color3
+
+type IColor3Shades = IColor2Shades | "a100" | "a200" | "a400" | "a100";
```

Comment:
type IColor3Shades = IColor2Shades | "a100" | "a200" | "a400" | "a700";

the last string should be "a700", not "a100".
a100 is duplicated.

Reply:
thanks for the review! fixed 👍